Спринт 10

rifnurV opened the pull request:
Создал два класса-контроллера. 
FilmController будет обслуживать фильмы, а UserController — пользователей.

Добавьте два класса: Film и User. Это классы — модели данных приложения. 
Добавил валидацию

rifnurV committed:
Создал два класса-контроллера. FilmController будет обслуживать фильмы, а UserController — пользователей.
Добавьте два класса: Film и User. Это классы — модели данных приложения.
Добавил валидацию

rifnurV committed:
CheckStyle

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/annotation/After1895ReleaseDate.java:
```diff
@@ -0,0 +1,21 @@
+@Constraint(validatedBy = After1895ReleaseDateValidator.class)
+@Target({ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+
```
Лишняя строчка

rifnurV replied:
убрал пустую строку

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,62 @@
+    }
+
+    // вспомогательный метод для генерации идентификатора нового поста
+    private long getNextId() {
```
Можно лучше
Можно завести отдельный счетчик для id, вместо того чтобы каждый раз высчитывать след. id (также, как было в таск-менеджере), или можете оставить этот вариант.

rifnurV replied:
упростил

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/model/User.java:
```diff
@@ -0,0 +1,29 @@
+import java.time.LocalDate;
+
+@Getter
+@Setter
```
Можно лучше
Вместо 
```
@Getter
@Setter
```
можно воспользоваться аннотацией ` @Data` . Эта аннотация помимо геттеров и сеттеров сгенерирует toString, equals и hashcode

rifnurV replied:
заменил

kesch9 replied:
У Film не подправили

rifnurV replied:
Верно оставил. Поправил

kesch9 commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,59 @@
+            log.info("Пользователь {} обновлен", newUser);
+            return newUser;
+        } else {
+            log.warn("Пользователь с id = " + newUser.getId() + " не найден", newUser.getId());
```
Не верно воспользовались форматированной строкой, надо не конкатенировать строки, а добавить {}
`log.warn("Пользователь с id = {} не найден", newUser.getId());`

rifnurV replied:
поправил.

kesch9 reviewed (commented):
Несколько комментариев к работе

rifnurV committed:
В аннотации убрал пустое поле
Упростил подсчет getNextId()
Для User @Getter @Setter заменил на @Data
Поправил опечатку в логе

kesch9 reviewed (approved):
Approve

rifnurV committed:
Film обавил @Data удалил @Getter @Setter
